### DISCUSSION TITLE
geting checksum mismatch issue

### DISCUSSION THREAD

**Discussion by vivekjaiswal90** [JeffyCN/meta-rockchip#95] [General]
Hello, I followed the instruction "https://github.com/JeffyCN/meta-rockchip/tree/kirkstone?tab=readme-ov-file"
I am using  the following machine configuration
 
`KERNEL_DEVICETREE = "rockchip/rk3399-khadas-edge-v.dtb"
UBOOT_MACHINE = "rk3399_defconfig"
RK_WIFIBT_FIRMWARES = " \
    rkwifibt-firmware-ap6354-wifi \
    rkwifibt-firmware-ap6354-bt \
    brcm-tools \
"`

But ia m alwasy getting the following error that checksum is not matching.
ALso i cannot figure out why the kernel version is 4.19. I am using the poky with kirkstone its latest and should be 5.15 

`vjaiswal@de-c-425:~/dev/Projects/SOLAR/khadas-dev/forTesting/yocto$ bitbake core-image-minimal
Loading cache: 100% |##########################################################################################################################################################################################################################| Time: 0:00:00
Loaded 2809 entries from dependency cache.
Parsing recipes: 100% |########################################################################################################################################################################################################################| Time: 0:00:00
Parsing of 1772 .bb files complete (1766 cached, 6 parsed). 2815 targets, 108 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "2.0.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-poky-linux"
MACHINE              = "rockchip-rk3399-khadas-edge-v"
DISTRO               = "poky"
DISTRO_VERSION       = "4.0.16"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta-rockchip        = "kirkstone:941f4102d94c8943bcbd6a12257605316fba7638"
meta                 
meta-poky            
meta-yocto-bsp       = "kirkstone:b3e316e8486e5462c3c71a9c8248779a5c253385"
meta-oe              = "kirkstone:fda737ec0cc1d2a5217548a560074a8e4d5ec580"

Initialising tasks: 100% |#####################################################################################################################################################################################################################| Time: 0:00:00
Sstate summary: Wanted 392 Local 0 Mirrors 0 Missed 392 Current 982 (0% match, 71% complete)
NOTE: Executing Tasks
ERROR: linux-rockchip-4.19-r0 do_populate_lic: QA Issue: linux-rockchip: The LIC_FILES_CHKSUM does not match for file://COPYING;md5=bbea815ee2795b2f4230826c0c6b8814
linux-rockchip: The new md5 checksum is 6bc538ed5bd9a7fc9398086aedcd7e46
linux-rockchip: Here is the selected license text:
vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
The Linux Kernel is provided under: 

SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note

Being under the terms of the GNU General Public License version 2 only,
according with:
    LICENSES/preferred/GPL-2.0

With an explicit syscall exception, as stated at:
..
LICENSES/exceptions/Linux-syscall-note

In addition, other licenses may also apply. Please see:

Documentation/process/license-rules.rst

for more details.

All contributions to the Linux Kernel are subject to this COPYING file.
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
linux-rockchip: Check if the license information has changed in /home/vjaiswal/dev/Projects/SOLAR/khadas-dev/forTesting/yocto/build/tmp/work-shared/rockchip-rk3399-khadas-edge-v/kernel-source/COPYING to verify that the LICENSE value "GPL-2.0-only" remains valid [license-checksum]
ERROR: linux-rockchip-4.19-r0 do_populate_lic: Fatal QA errors were found, failing task.
ERROR: Logfile of failure stored in: /home/vjaiswal/dev/Projects/SOLAR/khadas-dev/forTesting/yocto/build/tmp/work/rockchip_rk3399_khadas_edge_v-poky-linux/linux-rockchip/4.19-r0/temp/log.do_populate_lic.360139
ERROR: Task (/home/vjaiswal/dev/Projects/SOLAR/khadas-dev/forTesting/yocto/meta-rockchip/recipes-kernel/linux/linux-rockchip_4.19.bb:do_populate_lic) failed with exit code '1'
NOTE: Tasks Summary: Attempted 2629 tasks of which 2500 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/vjaiswal/dev/Projects/SOLAR/khadas-dev/forTesting/yocto/meta-rockchip/recipes-kernel/linux/linux-rockchip_4.19.bb:do_populate_lic
Summary: There were 2 ERROR messages, returning a non-zero exit code.
`

**Comment by JeffyCN** [comment]
That is weird, this part of code is unchanged for years.

1/ This layer is mainly focused on Rockchip's BSP packages and kernels, so it's actually using the Rockchip's BSP kernel 4.4/4.19/5.10/6.1.

2/ This "COPYING" file looks like to be a new version(from newer kernels),  not the expected 4.19 one which should come from:
https://github.com/JeffyCN/mirrors/blob/kernel-4.19-2022_11_23/COPYING
(Check the last line of it)

Guessing either your cached sources are dirty somehow, needs a clean rebuild(remove the packages/), or the kernel's source been overridden somewhere(check the local config)

**Reply by vivekjaiswal90** [reply]
I was able to get rid of the build error, by changing  this file https://github.com/JeffyCN/meta-rockchip/blob/master/conf/machine/include/rk3399.inc 

Changed from 4.19 to 5.10

```
PREFERRED_VERSION_linux-rockchip := "5.10%"
LINUXLIBCVERSION := "5.10-custom%"
```

Since I added a new machine and following as my kernel & uboot config
```
KERNEL_DEVICETREE = "rockchip/rk3399-khadas-edge-v.dtb"
UBOOT_MACHINE = "rk3399_defconfig"
```
